src/common/cockpitpipe.c

```diff
@@ -427,6 +448,31 @@ dispatch_error (gint fd,
   return TRUE;
 }
 
+static gboolean
+fd_readable (int fd)
+{
+  fd_set rfds;
+  struct timeval tv;
+
+  FD_ZERO (&rfds);
+  FD_SET (fd, &rfds);
+
+  tv.tv_sec = 0;
+  tv.tv_usec = 0;
+
+  return select (1, &rfds, NULL, NULL, &tv) > 0;
```

Comment:
```suggestion
  struct pollfd fd = { .fd = fd, .events = POLLIN };

  return poll (&fd, 1, 0) > 0;
```

Reply:
Pushed with two fixes (missing #include, and not duplicating the `fd` as both the function argument and the struct  variable name).